Framework/Algorithms/test/PolarizationEfficiencyCorTest.h

```diff
@@ -502,61 +400,16 @@ class PolarizationEfficiencyCorTest : public CxxTest::TestSuite {
       }
       return ws;
     } else if (kind == "histo") {
-      auto ws1 = createHistoWS(10, 0, 10);
-      auto ws2 = createHistoWS(10, 0, 10);
-      auto ws3 = createHistoWS(10, 0, 10);
-      auto ws4 = createHistoWS(10, 0, 10);
-
-      auto alg = AlgorithmFactory::Instance().create("JoinISISPolarizationEfficiencies", -1);
-      alg->initialize();
-      alg->setChild(true);
-      alg->setRethrows(true);
-      alg->setProperty("P1", ws1);
-      alg->setProperty("P2", ws2);
-      alg->setProperty("F1", ws3);
-      alg->setProperty("F2", ws4);
-      alg->setPropertyValue("OutputWorkspace", "dummy");
-      alg->execute();
-      MatrixWorkspace_sptr outWS = alg->getProperty("OutputWorkspace");
+      auto outWS = createEfficiencyOutputWorkspace(10, 10, "histo");
```

Comment:
I think ideally it would be good to pass the `kind` variable into `createEfficiencyOutputWorkspace` so that we don't create a new string. Perhaps we could simplify these two methods to be:

```
  MatrixWorkspace_sptr createEfficiencyOutputWorkspace(std::string const &kind) {
    const auto numWorkspaces = 4;
    const double endX = 10;

    std::vector<MatrixWorkspace_sptr> workspaces;
    if (kind == "histo") {
      for (int i = 0; i < numWorkspaces; ++i) {
        workspaces.push_back(createHistoWS(10, 0, endX));
      }
    } else if (kind == "points") {
      for (int i = 0; i < numWorkspaces; ++i) {
        workspaces.push_back(createPointWS(10, 0, endX));
      }
    } else if (kind == "points-short") {
      for (int i = 0; i < numWorkspaces; ++i) {
        workspaces.push_back(createPointWS(4, 0, endX));
      }
    } else {
      throw std::logic_error("Unknown efficiency test kind");
    }

    auto alg = AlgorithmFactory::Instance().create("JoinISISPolarizationEfficiencies", -1);
    alg->initialize();
    alg->setChild(true);
    alg->setRethrows(true);
    alg->setProperty("P1", workspaces[0]);
    alg->setProperty("P2", workspaces[1]);
    alg->setProperty("F1", workspaces[2]);
    alg->setProperty("F2", workspaces[3]);
    alg->setPropertyValue("OutputWorkspace", OUTPUT_GRP_NAME);
    alg->execute();

    return alg->getProperty("OutputWorkspace");
  }

  MatrixWorkspace_sptr createEfficiencies(std::string const &kind) {
    static std::map<std::string, std::vector<std::string>> const labels = {
        {WILDES_METHOD, {"P1", "P2", "F1", "F2"}}, {FREDRIKZE_METHOD, {"Pp", "Ap", "Rho", "Alpha"}}};
    if (kind == WILDES_METHOD || kind == FREDRIKZE_METHOD) {
      auto inWS = createWorkspaces(1)[0];
      MatrixWorkspace_sptr ws = WorkspaceFactory::Instance().create(inWS, 4);
      ws->getAxis(0)->setUnit("Wavelength");
      auto axis1 = std::make_unique<TextAxis>(4);
      auto axis1Raw = axis1.get();
      ws->replaceAxis(1, std::move(axis1));
      auto const &current_labels = labels.at(kind);
      for (size_t i = 0; i < ws->getNumberHistograms(); ++i) {
        axis1Raw->setLabel(i, current_labels[i]);
      }
      return ws;
    }

    return createEfficiencyOutputWorkspace(kind);
  }
```